matter_server/server/device_controller.py

```diff
@@ -230,6 +230,8 @@ async def stop(self) -> None:
             scan_task.cancel()
         if self._aiozc:
             await self._aiozc.async_close()
+        for task in self._setup_node_with_retry_tasks.values():
+            task.cancel()
```

Comment:
```suggestion
        for task in self._setup_node_with_retry_tasks.values():
            if not task.done():
                task.cancel()
```

Reply:
`task.cancel()` is safe to be called for cancelled tasks. That said, maybe we should remove the tasks once they are done? :thinking:  On the other hand, why bother...

Reply:
I was just wondering the same, should we cleanup the list and pop an item out if its done.